server/src/main/java/org/elasticsearch/common/document/DocumentField.java

```diff
@@ -132,7 +133,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
-    public static DocumentField fromXContent(XContentParser parser) throws IOException {
+    public static DocumentField fromXContent(XContentParser parser, boolean inMetadataArea) throws IOException {
```

Comment:
`inMetadataArea` why not call this parameter the same `isMetadataField`?

Reply:
this parameter naming follows the intent to emphasize and make it explicit that we decide whether the field is metadata or not depending on whether it is located in metadata area of xcontent or not. 

I don't really have a strong opinion here, we can replace it with isMetadataField and add some comments, if you prefer that.

Reply:
thanks for the explanation, `isMetadataField` sounds better for consistency with the rest of the code. And making a field being a part of medata makes it a metadata field.